carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+           }
+            
+        message->value.choice.TestMessage00.body.planType=type;
+
+        //urgency
+        message->value.choice.TestMessage00.body.urgency=plainMessage.urgency;
+
+        //location
+        long location_val;
+        location_val=plainMessage.location.ecef_x;
+        if(location_val> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage00.body.location.ecefX=location_val;
+
+        location_val=plainMessage.location.ecef_y;
+        if(location_val> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefY is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage00.body.location.ecefY=location_val;
+
+        location_val=plainMessage.location.ecef_z;
+        if(location_val> LOCATION_MAX_Z || location_val<LOCATION_MIN_Z){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage00.body.location.ecefZ=location_val;
+
+        uint64_t location_time=plainMessage.location.timestamp;
+        std::string location_timestamp=std::to_string(location_time);
+        size_t location_timestamp_string_size=location_timestamp.size();
+        uint8_t string_content_location_timestamp[location_timestamp_string_size];
+        for(size_t i=0;i<location_timestamp_string_size;i++)
+        {
+            string_content_location_timestamp[i]=location_timestamp[i];
+        }
+        message->value.choice.TestMessage00.body.location.timestamp.buf=string_content_location_timestamp;
+        message->value.choice.TestMessage00.body.location.timestamp.size=location_timestamp_string_size;
+
+        //strategyParams
+        size_t params_string_size=plainMessage.strategy_params.size();
+        uint8_t string_content_params[params_string_size];
+        for(size_t i=0;i<params_string_size;i++)
+        {
+            string_content_params[i]=plainMessage.strategy_params[i];
+        }
+        message->value.choice.TestMessage00.body.strategyParams.buf=string_content_params;
+        message->value.choice.TestMessage00.body.strategyParams.size=params_string_size;
+        
+        //Trajectory
+            //trajectoryStart
+        MobilityLocation* trajectory_location;
+        trajectory_location=(MobilityLocation*)calloc(1,sizeof(MobilityLocation));
+        if(!trajectory_location)
+        {
+            ROS_WARN_STREAM("Cannot allocate mem for trajectory.location encoding");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        long trajectory_start;
+        trajectory_start=plainMessage.trajectory.location.ecef_x;
+        if(trajectory_start> LOCATION_MAX || location_val<LOCATION_MIN){
```

Comment:
Is the second part here supposed to be `|| trajectory_start<LOCATION_MIN` not `location_val<...`?

Reply:
Yes, you're right, it should be trajectory_start